python/ctsm/modify_fsurdat/modify_fsurdat.py

```diff
@@ -0,0 +1,288 @@
+        for var_name, var in vars_4d.items():
+            if var is not None:
+                self.set_lai_sai_hgts(dom_nat_pft=dom_nat_pft,
+                                      var_name=var_name, var=var)
+
+
+    def set_lai_sai_hgts(self, dom_nat_pft, var_name, var):
+        """
+        Description
+        -----------
+        If user has specified lai, sai, hgt_top, hgt_bot, replace these with
+        values selected by the user for dom_nat_pft. Else do nothing.
+        """
+        if dom_nat_pft == 0:  # bare soil: var must equal 0
+            var = [0] * 12
+        if len(var) != 12:
+            errmsg = 'Error: Variable should have exactly 12 ' \
+                     'entries in the configure file: ' + var_name
+            abort(errmsg)
+        for mon in self.file.time - 1:  # loop over 12 months
+            # set 4D variable to value for dom_nat_pft
+            self.file[var_name][mon,dom_nat_pft,:,:] = self.setvar(
+                self.file[var_name][mon,dom_nat_pft,:,:], var[int(mon)])
+
+
+    def zero_nonveg(self):
+        """
+        Description
+        -----------
+        Set all landunit weights to 0 except the natural vegetation landunit.
+        Set that one to 100%.
+
+        Arguments
+        ---------
+        """
+
+        self.file['PCT_NATVEG'][:,:] = 100
+        self.file['PCT_CROP'][:,:] = 0
+        self.file['PCT_LAKE'][:,:] = 0
+        self.file['PCT_WETLAND'][:,:] = 0
+        self.file['PCT_URBAN'][:,:] = 0
+        self.file['PCT_GLACIER'][:,:] = 0
+
+
+    def setvar(self, var, val):
+        """
+        Set variable var to value val in the *rectangle* domain, which
+        is denoted as *other* in the *not_rectangle* domain.
+        """
+
+        var = var.where(self.not_rectangle, other=val)
+
+        return var
+
+
+    def set_idealized(self):
+        """
+        Description
+        -----------
+        Set fsurdat variables in a rectangle defined by lon/lat limits
+        """
+
+        # Overwrite in rectangle(s)
+        # ------------------------
+        # If idealized, the user makes changes to variables as follows.
+        # "other" assigns the corresponding value in the rectangle.
+        # Values outside the rectangle are preserved.
+        # ------------------------
+
+        # Default values
+        zbedrock = 10
+        max_sat_area = 0  # max saturated area
+        max_inundated = 0  # max inundated fraction
+        std_elev = 0  # standard deviation of elevation
+        slope = 0  # mean topographic slope
+        pftdata_mask = 1
+        landfrac_pft = 1
+        # if pct_nat_veg had to be set to less than 100, then each special
+        # landunit would have to receive a unique pct value rather than the
+        # common value used here in pct_not_nat_veg = 0
+        pct_nat_veg = 100  # do not change; works with pct_not_nat_veg = 0
+        pct_not_nat_veg = 0  # do not change; works with pct_nat_veg = 0
```

Comment:
- [ ] Thanks for the above comments; this one should say "works with pct_nat_veg = 100" (not 0)

Reply:
thank you for catching